File: go/15_summation.go
package main

import "fmt"

// https://www.codewars.com/kata/55d24f55d7dd296eb9000030/train/go

// func Summation(n int) int {
// 	var a int
// 	for i := n; i >= 1; i-- {
// 		a += i
// 	}

// 	return a
// }

func Summation(n int) int {
	return n * (n + 1) / 2
}

func main() {
	fmt.Println(Summation(8)) // 36
}

File: js/_/15_summation.js
// https://www.codewars.com/kata/55d24f55d7dd296eb9000030/train/javascript

function summation(num) {
  let result = 0;
  for (a = num; a >= 1; a--) {
    result += a;
  }

  return result;
}

console.log(summation(1)) // 1
console.log(summation(2)) // 3
console.log(summation(8)) // 36